starter-code/src/ContactOverview.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react';
+
+import contacts from './contacts.json';
+
+import ContactTable from './ContactTable.js';
+import ContactRow from './ContactRow.js';
+
+class ContactOverview extends Component {
+  state = {
+    currentContacts: contacts.slice(0, 5)
+  }
+
+  createRandomContact = () => {
+    let unChosenContacts = this.getUnChosenContacts()
+    let randomContact = unChosenContacts[Math.floor(Math.random() * unChosenContacts.length)];
+    this.setState({ currentContacts: [...this.state.currentContacts, randomContact] })
+  }
+
+  getUnChosenContacts = () => {
+    return contacts.filter(cont => {
+      return !this.state.currentContacts.includes(cont);
+    })
+  }
+
+  sortContactByName = () => {
+    let sortedArray = [...this.state.currentContacts].sort((contactOne, contactTwo) => {
+      var nameA = contactOne.name.toUpperCase();
+      var nameB = contactTwo.name.toUpperCase();
+      return (nameA < nameB) ? -1 : (nameA > nameB) ? 1 : 0;
+    });
+    this.setState({ currentContacts: sortedArray });
+  }
+
+  sortContactByPopularity = () => {
+    let sortedArray = [...this.state.currentContacts].sort((contactOne, contactTwo) => {
+      return (contactOne.popularity < contactTwo.popularity) ? -1 : (contactOne.popularity > contactTwo.popularity) ? 1 : 0;
+    });
+    this.setState({ currentContacts: sortedArray });
+  }
+
+  hideContact = (name) => {
+    // find in currentContacts
+    let selectedContact = this.state.currentContacts.map(cont => {
+      return cont.name;
+    }).indexOf(name);
+    // delete from array and update state
+    let newContacts = this.state.currentContacts.splice(selectedContact, 1)
+    this.setState({ currentContact: newContacts })
+  }
+
+  render() {
+    return (
+      <div>
+        <h1>{this.props.title}</h1>
+
+        <p className="button-action" onClick={this.createRandomContact}>Add random contact</p>
```

Comment:
A small tip would be, to move all buttons to a separate component called for example `<TableControls/>`. when you want to extend you a table with new controls you can add all that logic to the `<TableControls/>` component and use your `<ContactOverview/>` component as a component that constructs the overview from separate components.